File: proj.android/src/com/dxco/crimson/SoundUtil.java
package com.dxco.crimson;

import android.app.Activity;

public class SoundUtil {

	public static Activity activity;

	public static boolean isMusicOn() {
		return ConfigurationUtil.getConfiguration(activity, "isMusicOn", true);
	}
	
	public static boolean isSoundEffectsOn() {
		return ConfigurationUtil.getConfiguration(activity, "isSoundEffectsOn", true);
	}
	
	public static void enableMusic() {
		ConfigurationUtil.editConfiguration(activity, "isMusicOn", true);
	}
	
	public static void disableMusic() {
		ConfigurationUtil.editConfiguration(activity, "isMusicOn", false);
	}
	
	public static void enableSoundEffects() {
		ConfigurationUtil.editConfiguration(activity, "isSoundEffectsOn", true);
	}
	
	public static void disableMusicEffects() {
		ConfigurationUtil.editConfiguration(activity, "isSoundEffectsOn", false);
	}

	public static int isSoundEffectsCpp() {
		return isSoundEffectsOn() ? 1 : 0;
	}
	
	public static int isMusicOnCpp() {
		return isMusicOn() ? 1 : 0;
	}
}

File: Classes/crimson/SoundUtil.cpp
#include "SoundUtil.h"
#include "cocos2d.h"

#include "platform/android/jni/JniHelper.h"

#include <jni.h>


namespace dxco {

bool SoundUtil::isMusicOn() {

	bool result = false;

	cocos2d::JniMethodInfo t;

	if (cocos2d::JniHelper::getStaticMethodInfo(t, "com/dxco/pandavszombies/SoundUtil", "isMusicOnCpp", "()I")) {

		int ret = (int) t.env->CallStaticIntMethod(t.classID, t.methodID);

		if (ret) {
			result = true;
		}
	}

	return result;
}

bool SoundUtil::isSoundEffectsOn() {

	bool result = false;

	cocos2d::JniMethodInfo t;

	if (cocos2d::JniHelper::getStaticMethodInfo(t, "com/dxco/pandavszombies/SoundUtil", "isSoundEffectsCpp", "()I")) {

		int ret = (int) t.env->CallStaticIntMethod(t.classID, t.methodID);

		if (ret) {
			result = true;
		}
	}

	return result;
}

} /* namespace dxco */
